DeviceDetector.php

```diff
@@ -590,14 +611,18 @@ protected function parseBot()
             return false;
         }
 
-        $botParser = new Bot();
-        $botParser->setUserAgent($this->getUserAgent());
-        $botParser->setYamlParser($this->getYamlParser());
-        $botParser->setCache($this->getCache());
-        if ($this->discardBotInformation) {
-            $botParser->discardDetails();
```

Comment:
This removes the special `discardDetails` for the bot parser. This is quite important, as it speeds up the detection for bots a lot if no detailed bot information is needed.

Reply:
Ow, that's really bad, sorry for that! I'll write a fix for that tomorrow, tried to make it too much te same as the other 2 functions, and then I stopped paying attention. 

Are you open for the concept of this MR though?

Reply:
Sure, why not allowing multiple / custom bot parsers

Reply:
Added a test that picks up this bug (again, that was careless, sorry), and added a fix. Needed to add an abstract class to define the `discardDetails()` method for the bots.